shopfloor/actions/completion_info.py

```diff
@@ -0,0 +1,42 @@
+from odoo import _
+
+from odoo.addons.component.core import Component
+
+
+class CompletionInfo(Component):
+    """Provide methods for completion info of pickings
+
+    They are based on the module "stock_picking_completion_info" from
+    OCA/stock-logistics-warehouse.
+    """
+
+    _name = "shopfloor.completion.info.action"
+    _inherit = "shopfloor.process.action"
+    _usage = "completion.info"
+
+    def popup(self, move_lines):
+        """Return a popup if move lines make chained pickings ready
+
+        Return None in case no popup should be displayed.
+        """
+        pickings = move_lines.mapped("picking_id").filtered(
+            lambda p: p.picking_type_id.display_completion_info
+            and p.completion_info == "next_picking_ready"
+        )
+        if not pickings:
+            return None
+        next_pickings = pickings.mapped("move_lines.move_dest_ids.picking_id").filtered(
+            lambda p: p.state == "assigned"
+        )
+        if not next_pickings:
+            return None
+        return {
+            "buttons": "ok",
```

Comment:
what is `buttons` for?

Reply:
We might want to add "ok_back" to have a confirmation screen (when the confirmation screen does not need a view with data but only a message). But maybe I should remove this until we have the use case.

Reply:
for now we assume that we have only one button "ok" that will just close the message.
If we have to support more actions I'd rather use an `actions` list or dicts, but this can be tricky so let's leave it behind for now.